src/sentry/services/hybrid_cloud/user/impl.py

```diff
@@ -27,10 +32,24 @@
 )
 
 
-class DatabaseBackedUserService(
-    FilterQueryDatabaseImpl[User, UserFilterArgs, RpcUser, UserSerializeType],
-    UserService,
-):
+class DatabaseBackedUserService(UserService):
+    @cached_property
+    def _fq(self) -> _UserFilterQuery:
+        return self._UserFilterQuery()
```

Comment:
Are filter queries mutated in place? If they are mutated this could leak state between queries.

Reply:
No, it's stateless. This should be safe.